zp-relayer/workers/sentTxWorker.ts

```diff
@@ -124,81 +80,92 @@ export async function createSentTxWorker<T extends EstimationType>(gasPrice: Gas
+        for (let wj of waitingJobs) {
+          // One of the jobs can be undefined, so we need to check it
+          // https://github.com/taskforcesh/bullmq/blob/master/src/commands/addJob-8.lua#L142-L143
+          if (!wj?.id) continue
+          waitingJobIds.push(wj.id)
+          reschedulePromises.push(poolTxQueue.add(txHash, [wj.data.txPayload]).then(j => j.id as string))
+        }
+        logger.info('Marking ids %j as failed', waitingJobIds)
+        await markFailed(redis, waitingJobIds)
+        logger.info('%s Rescheduling %d jobs to process...', logPrefix, waitingJobs.length)
+        const rescheduledIds = await Promise.all(reschedulePromises)
+
+        return [SentTxState.REVERT, txHash, rescheduledIds] as SentTxResult
       }
     } else {
       // Resend with updated gas price
       const txConfig = job.data.txConfig
 
       const oldGasPrice = job.data.gasPriceOptions
-      const fetchedGasPrice = gasPrice.getPrice()
 
-      const newGasPrice = chooseGasPriceOptions(oldGasPrice, fetchedGasPrice)
+      const fetchedGasPrice = await gasPrice.fetchOnce()
+      const oldWithExtra = addExtraGasPrice(oldGasPrice, config.minGasPriceBumpFactor, null)
+      const newWithExtra = addExtraGasPrice(fetchedGasPrice, config.gasPriceSurplus, null)
 
-      logger.warn('Tx %s is not mined; updating gasPrice: %o -> %o', txHash, oldGasPrice, newGasPrice)
+      const newGasPrice = chooseGasPriceOptions(oldWithExtra, newWithExtra)
+
+      logger.warn('%s Tx %s is not mined; updating gasPrice: %o -> %o', logPrefix, txHash, oldGasPrice, newGasPrice)
 
       const newTxConfig = {
         ...txConfig,
         ...newGasPrice,
       }
 
       try {
-        if (txType === TxType.PERMITTABLE_DEPOSIT) {
-          const deadline = (txData as PermittableDepositTxData).deadline
-          await checkAssertion(() => checkDeadline(toBN(deadline), config.permitDeadlineThresholdResend))
-        }
-
         const newTxHash = await signAndSend(newTxConfig, config.relayerPrivateKey, web3)
-
-        // Add updated job
-        await sentTxQueue.add(
-          newTxHash,
-          {
-            ...job.data,
-            txHash: newTxHash,
-            txConfig: newTxConfig,
-            gasPriceOptions: newGasPrice,
-          },
-          {
-            priority: txConfig.nonce,
-            delay: config.sentTxDelay,
-          }
-        )
-        return [SentTxState.RESEND, newTxHash] as SentTxResult
+        // Update job
+        await job.update({
+          ...job.data,
+          txHash: newTxHash,
+          txConfig: newTxConfig,
+          gasPriceOptions: newGasPrice,
+        })
+        await job.updateProgress({ txHash: newTxHash, gasPrice: newGasPrice })
       } catch (e) {
         const err = e as Error
-        logger.warn('%s: Tx resend failed for %s: %s', logPrefix, txHash, err.message)
-        if (isSameTransactionError(err) || isGasPriceError(err)) {
```

Comment:
`isSameTransactionError` should not really happen now, but let's treat it as if sending was successful.

`isGasPriceError` will mean that relayer gas bump politics is improperly configured and will lead to a re-send deadlock, right?

Reply:
> `isSameTransactionError` should not really happen now, but let's treat it as if sending was successful.

Now, in case of such error, another iteration of re-send loop will be triggered, so I don't think we should handle this error
> `isGasPriceError` will mean that relayer gas bump politics is improperly configured and will lead to a re-send deadlock, right?

Agree, probably we should still update job's `gasPrice` in this case, so that new value will be used in next iteration of re-send loop.

Reply:
Btw, any thrown error in the processor function will trigger next iteration of re-send loop